61_rotate_ist.py

```diff
@@ -0,0 +1,30 @@
+# Definition for singly-linked list.
+# class ListNode:
+#     def __init__(self, val=0, next=None):
+#         self.val = val
+#         self.next = next
+class Solution:
+    def rotateRight(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
+        if not head: return head
+        dummy = behind = ahead = head
+        N = 0
+        
+        while dummy:
+            dummy = dummy.next
+            N += 1
+            
+        k = N - k % N
```

Comment:
You could use more descriptive variable names. Except for these, the other naming's are good.